app/app.py

```diff
@@ -1,167 +1,113 @@
-from tensorflow.keras.models import load_model
-import joblib
+from transformers import BertTokenizer, BertForSequenceClassification
+from src.api.security import authenticate_user
+from datetime import datetime, timedelta
+
+
+# Load configuration
+with open(os.path.join('configs', 'config.yaml'), 'r') as config_file:
+    config = yaml.safe_load(config_file)
 
 load_dotenv()
-# JWT settings
-SECRET_KEY = os.getenv("SECRET_KEY", "your_secret_key")
+
+SECRET_KEY = os.getenv("SECRET_KEY")
+
 ALGORITHM = "HS256"
 ACCESS_TOKEN_EXPIRE_MINUTES = 30
 
-# Initialize FastAPI app
+# Initialize FastAPI
 app = FastAPI()
-logger = get_logger('api')
 
-pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+# Load model and tokenizer
+model_path = os.path.join(config['model']['output_dir'], 'bert_classifier.pth')
+model = BertForSequenceClassification.from_pretrained(config['model']['bert_model_name'], num_labels=config['model']['num_classes'])
+model.load_state_dict(torch.load(model_path, map_location=torch.device('cpu')))
+model.eval()
 
-# OAuth2 scheme
-oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+tokenizer = BertTokenizer.from_pretrained(config['model']['bert_model_name'])
 
-# Load model and tokenizer
-model = load_model('models/lstm_model.h5')
-tokenizer = joblib.load('models/tokenizer.pkl')
-
-# In-memory user database (for demonstration purposes)
-users_db = {
-    "user1": {
-        "username": "user1",
-        "full_name": "User One",
-        "password": pwd_context.hash("user1password"),
-        "role": "user",
-    },
-    "admin": {
-        "username": "admin",
-        "full_name": "Admin User",
-        "password": pwd_context.hash("adminpassword"),
-        "role": "admin",
-    },
-}
-
-# Pydantic models
-class Review(BaseModel):
-    text: str
-
-# Helper functions
-def verify_password(plain_password, hashed_password):
-    return pwd_context.verify(plain_password, hashed_password)
-
-def authenticate_user(username: str, password: str):
-    user = users_db.get(username)
-    if not user or not verify_password(password, user["password"]):
-        return None
-    return user
-
-def create_access_token(data: dict, expires_delta: Union[timedelta, None] = None):
+# Logging setup
+logging.basicConfig(filename='api_requests.log', level=logging.INFO, format='%(asctime)s - %(message)s')
+
+class ReviewInput(BaseModel):
+    review: str
+
+class TokenData(BaseModel):
+    username: str
+
+def create_access_token(data: dict, expires_delta: timedelta = None):
     to_encode = data.copy()
     if expires_delta:
-        expire = datetime.now(timezone.utc) + expires_delta
+        expire = datetime.utcnow() + expires_delta
     else:
-        expire = datetime.now(timezone.utc) + timedelta(minutes=15)
+        expire = datetime.utcnow() + timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
     to_encode.update({"exp": expire})
     encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
     return encoded_jwt
```

Comment:
Here regarding the authentication mechanism that currently we are using, I changed to be consistent with the first project, and its currently pushed in the `Database PR`. I think we will have to work on the merge of our `PRs` together. I think we will have conflicts because we both changed a lot in this file `app.py`

Reply:
Yes, I didn't change much actually. The main thing I did was that I replace the code implementation of the previous model with the current one (the bert model).